core/project_settings.cpp

```diff
@@ -284,6 +284,17 @@ void ProjectSettings::_convert_to_last_version(int p_from_version) {
 			}
 		}
 	}
+
+	// Deletes invalid global script entries from project settings.
```

Comment:
I don't think this is needed, `editor/editor_sectioned_inspector.cpp` already filters out all properties starting with `global/`.